src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -141,6 +152,19 @@ const NavSearchBar = () => {
     [searchQuery]
   );
 
+  const inputRef = useRef<HTMLInputElement>(null);
+  const focusSearchBar = useCallback((e) => {
+    e.preventDefault();
+    if (inputRef && inputRef.current) inputRef.current.focus();
+  }, []);
```

Comment:
This will have the same effect, though in shorter/clearer code (using TypeScript's 'optional chaining' feature):

```suggestion
  const inputRef = useRef<HTMLInputElement>(null);
  const focusSearchBar = useCallback((e) => {
    e.preventDefault();
    inputRef.current?.focus();
  }, []);
```

Reply:
@mike-lvov @sunny-viktoryia I know we've talked a bit about avoiding using `e.preventDefault()` of late, though still don't fully have my head around the 'what/when' of it yet.

Do you have any thoughts/suggestions on whether we need to use it here for this, and if so, what our 'preferred' way of solving this is?

Reply:
(this might actually be the pattern we use in all of our keybinding handlers, based on a comment in `useMapKeyboardControls`

Reply:
applied the above suggested changes in https://github.com/sparkletown/sparkle/pull/1416/commits/2dc066902d7fe56f50eb54583cc33294a7f246db

The `e.preventDefault()` part of this is still oustanding.

Reply:
I believe it was needed to avoid the key propagate into search input element. So the question remaining is either it should be moved to mousetrap handler itself?

Reply:
> I believe it was needed to avoid the key propagate into search input element.

According the the `mousetrap` docs, that shouldn't be required when we're in an `input` field and similar:

- https://craig.is/killing/mice#api.bind.text-fields
- https://craig.is/killing/mice#api.stopCallback

Reply:
I have triple checked that removing e.preventDefault pastes `/` into input.  If mousetrap is supposed, it doesn't (may be due to all the layering to get to that `input`).

Also there is a comment
```
// TODO: use e.preventDefault() or return false or similar in the keyboard handlers (check mousetrap docs) so we don't scroll on arrow key presses
```
for `useMapKeyboardControls` so observed needed elsewhere.  Not certain if digging deeper is productive on this one.

Reply:
> not even sure if that would be a good strategy from mouseTrap to just do it by default - I might need a callback to just act on the character but do pass it through to input -- who knows?!

Which is why they provide the documentation that explains how it works and how to change it if you need to, that I provided above in https://github.com/sparkletown/sparkle/pull/1416#discussion_r639442598

---

> I have triple checked that removing e.preventDefault pastes / into input. If mousetrap is supposed, it doesn't

Fair. Potentially because we are using global keybindings rather than directly on the element.

---

> Also there is a comment for useMapKeyboardControls so observed needed elsewhere

Yup, because movement on the map doesn't relate to `input` and similar things that `mousetrap` disables it for by default. That comment may also not even be strictly true, it was written quite 
a while ago (by me) when I had less understanding of how it all interacts.